lib/mpdecimal.ex

```diff
@@ -1,10 +1,19 @@
 defmodule MPDecimal do
   alias MPDecimal.Nif
 
-  def power(%Decimal{} = base, %Decimal{} = power) do
-    base = Decimal.to_string(base, :xsd)
-    power = Decimal.to_string(power, :xsd)
-    res = Nif.power(base, power)
-    Decimal.new(res)
+  def power(%Decimal{} = base, %Decimal{} = exponent) do
+    # Add the null terminator to the end of each string argument so that it
+    # forms a valid cstring for consumption by the NIF.
+    base = Decimal.to_string(base, :xsd) <> "\0"
+    exponent = Decimal.to_string(exponent, :xsd) <> "\0"
+    
+    case Nif.power(base, exponent) do
+      {:ok, result} -> Decimal.new(result)
+      {:error, message} -> raise(MPDecimal.Error, message: message)
```

Comment:
I would suggest that in the error case, we should return `{:error, MPDecimal.Error.exception(message: message)}`. And we could add a second function here called `power!(base, exponent)` which raises the exception in case of an error.

This allows the caller to choose whether they want to receive errors are return values, or let them be raised. This follows a convention that you can see in things like [File.read](https://github.com/elixir-lang/elixir/blob/v1.12.0/lib/elixir/lib/file.ex#L349)

Reply:
Great suggestion. I'll take a closer look at this convention and refactor accordingly.

Reply:
I think the rest of the `Decimal` library would be a better source of conventions than the `File` API: if you call `Decimal.div(42, 0)`, a `Decimal.Error` gets thrown. I don't think there are any functions in the `Decimal` library that follow the `:ok/:error` tuple pattern.

Personally, I don't see this as a problem; if math fails it's common across all of Elixir that that is an exceptional case that should be thrown/caught.

Reply:
That's a great point. Following the `Decimal` convention makes a lot of sense here. There were times that I *wished* the `Decimal` library would return `{:error, exception}` tuples while working on peg, but that is just our own use-case and I think consistency with the ecosystem is a bigger concern.

I vote we ship this as-it-is

Reply:
I had the same thoughts here. I was intentional about matching the behavior of the Decimal library, so it seems we are all in agreement.